docs/Unicode.md

```diff
@@ -0,0 +1,129 @@
+# Unicode
+
+This document details suggestions on how to handle unicode in Flux as well as how unicode is handled 
+in selected query languages.
+
+## Unicode Summary
+
+UTF-8 is defined to encode code points in 1 to 4 bytes. From Wikipedia, "The first 128 characters 
+(US-ASCII) need one byte. The next 1,920 characters need two bytes to encode, which covers the remainder 
+of almost all Latin-script alphabets, and also Greek, Cyrillic, Coptic, Armenian, Hebrew, Arabic, Syriac, 
+Thaana and N'Ko alphabets, as well as Combining Diacritical Marks. Three bytes are needed for characters 
+in the rest of the Basic Multilingual Plane, which contains virtually all characters in common use 
+including most Chinese, Japanese and Korean characters. Four bytes are needed for characters in the other 
+planes of Unicode, which include less common CJK characters, various historic scripts, mathematical 
+symbols, and emoji (pictographic symbols)."
+
+## Functions and Functionality in Flux affected by unicode
+
+### Length
+
+Should we define length as the number of characters or the number of bytes?
+(Will be the same for the 128 US-ASCII characters that we are used to, but will be different for other
+languages that Flux may want to support.)
+
+### Substring/Slicing/Indexing
+
+Will probably be more convenient to define length as number of values in the resulting array 
+when `strings.split(v: "...", t: "")` is applied. Alternatively, can define length of array 
+and length of strings differently?
+
+## How unicode is dealt with in other languages
+
+### Python
+
+Has `bytes.decode()` and `str.encode()` methods that translate unicode to string and vice versa. 
+One-character unicode strings can also be created with the chr() built-in function.
+
+`unicodedata.normalize()` converts strings to one of several normal forms, where letters followed
+ by a combining character are replaced with single characters. normalize() can be used to perform
+ string comparisons that won’t falsely report inequality if two strings use combining characters
+ differently.
+
+```
+import unicodedata
+
+def compare_strs(s1, s2):
+    def NFD(s):
+        return unicodedata.normalize('NFD', s)
+
+    return NFD(s1) == NFD(s2)
+
+single_char = 'ê'
+multiple_chars = '\N{LATIN SMALL LETTER E}\N{COMBINING CIRCUMFLEX ACCENT}'
+print('length of first string=', len(single_char))
+print('length of second string=', len(multiple_chars))
+print(compare_strs(single_char, multiple_chars))
+```
+Returns:
+```
+length of first string= 1
+length of second string= 2
+True
+```
+
+Documentation: https://docs.python.org/3/howto/unicode.html
+
+### Go
+
+Go encodes string as byte array and thus the `length` is determined by the number of bytes 
+instead of the number of characters.
+
+Documentation/Source:
+- https://golang.org/pkg/unicode/
+- https://coderwall.com/p/k7zvyg/dealing-with-unicode-in-go
```

Comment:
can you summarize this a bit?  I know go has something called a 'glyph'  is that what they call a unicode character? what functions are available for detecting and processing unicode characters in a string?   Is there a way to iterate over characters or isolate a single unicode char?

Reply:
Only glyphs I can find are Hieroglyphs that are part of the *Rangetables. 

There's the functions in the Unicode package and [utf8 pacakge](https://golang.org/pkg/unicode/utf8/). utf8 does provide ways to see how many bytes each character is. 

I'll add this to the doc.